### ISSUE TITLE
How to catch and redirect to 500 error in a crash on actix 2?

### ISSUE THREAD

**Issue by mamcx** [actix/actix-web#1504]
This is happening after I migrate to 2. When a crash happened (a unwarp gone wrong, for example), the  error handlers not catch it, the error backtrace is repeated * thread number, the browser answer with a connection reset:

> May 12 10:21:22.050 INFO Starting "actix-web-service-127.0.0.1:5000" service on 127.0.0.1:5000, version: 0.2.0
> thread 'actix-rt:worker:0' panicked at 'internal error: entered unreachable code', server/src/core.rs:92:9
> stack backtrace:
>    0:        0x1083a2cbf - backtrace::backtrace::libunwind::trace::h6428452ebf6513c0
>                                at /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/libunwind.rs:86
>    1:        0x1083a2cbf - backtrace::backtrace::trace_unsynchronized::h8bc3f3c9d1e99578
>                                at /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/mod.rs:66
> **.. a wall of errors here, repeated 6 times (I have 6 cores)**

```rust
fn error_page<B>(res: dev::ServiceResponse<B>) -> Result<ErrorHandlerResponse<B>> 

#[actix_rt::main]
async fn main() -> Result<(), CoreErr> {
    let _scope_guard = corelib::slog_scope::set_global_logger(LOG.clone());
    corelib::slog_stdlog::init_with_level(log::Level::Info).unwrap();
    let pool = Db::open_server(None)?;

    HttpServer::new(move || {
        App::new()
            .data(pool.clone())
            .wrap(middleware::NormalizePath)
            .wrap(middleware::Compress::default())
            .wrap(
                Cors::new()
                    .supports_credentials()
                    .max_age(3600)
                    .finish(),
            )
            .wrap(
                ErrorHandlers::new()
                    .handler(http::StatusCode::NOT_FOUND, error_page)
                    .handler(http::StatusCode::INTERNAL_SERVER_ERROR, error_page),
            )
            .wrap(IdentityService::new(
                CookieIdentityPolicy::new(...)
                    .name("auth")
                    .max_age(60 * 60 * 24) //1 dias
                    .secure(!is_debug()),
            ))
            .wrap(middleware::Logger::default()) //always last
            .service(web::resource("/login").to(admin::login))
       })
    .bind("127.0.0.1:5000")?
    .run()
    .await
    .map_err(|source| CoreErr::IOError { source })
}
```

**Comment by robjtede** [comment]
see recent discussion about it here:
https://github.com/actix/actix-web/issues/1501

tl;dr: handle your errors
